sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompositeOffset.scala

```diff
@@ -22,7 +22,7 @@ package org.apache.spark.sql.execution.streaming
  * [[Source]]s that are present in a streaming query. This is similar to simplified, single-instance
  * vector clock that must progress linearly forward.
  */
-case class CompositeOffset(offsets: Seq[Option[Offset]]) extends Offset {
+case class CompositeOffset(offsets: Array[Option[Offset]]) extends Offset {
```

Comment:
Change the type to `Array` as jackson-module-scala doesn't support `Seq[Option[X]]`. Probably because of the type erasure.